packages/aws-library/src/aws_library/ec2/_models.py

```diff
@@ -67,19 +67,26 @@ class EC2InstanceType:
 InstancePrivateDNSName: TypeAlias = str
 
 
-class AWSTagKey(ConstrainedStr):
-    # see [https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/Using_Tags.html#tag-restrictions]
-    regex = re.compile(r"^(?!(_index|\.{1,2})$)[a-zA-Z0-9\+\-=\._:@]+$")
-    min_length = 1
-    max_length = 128
-
-
-class AWSTagValue(ConstrainedStr):
-    # see [https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/Using_Tags.html#tag-restrictions]
-    # quotes []{} were added as it allows to json encode. it seems to be accepted as a value
-    regex = re.compile(r"^[a-zA-Z0-9\s\+\-=\.,_:/@\"\'\[\]\{\}]*$")
-    min_length = 0
-    max_length = 256
+# see [https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/Using_Tags.html#tag-restrictions]
+AWSTagKey: TypeAlias = Annotated[
+    str,
+    StringConstraints(
+        min_length=1,
+        max_length=128,
+        pattern=r"^(?!(_index|\.{1,2})$)[a-zA-Z0-9\+\-=\._:@]+$",
+    ),
+]
```

Comment:
In Pydantic v2, the pydantic-core library no longer supports advanced regex features such as look-ahead assertions (like ^(?!...)).